pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -264,3 +265,29 @@ func aggregateStatefulSetStatus(object *unstructured.Unstructured, aggregatedSta
 
 	return helper.ToUnstructured(statefulSet)
 }
+
+func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (*unstructured.Unstructured, error) {
+	pod, err := helper.ConvertToPod(object)
+	if err != nil {
+		return nil, err
+	}
+
+	newStatus := &corev1.PodStatus{}
+	for _, item := range aggregatedStatusItems {
+		if item.Status == nil {
+			continue
+		}
+		if err = json.Unmarshal(item.Status.Raw, newStatus); err != nil {
+			return nil, err
+		}
+		klog.V(3).Infof("Grab pod(%s/%s) status from cluster(%s), phase: %s, qosClass: %s", newStatus.Phase, newStatus.QOSClass)
+	}
```

Comment:
This logic looks like it only unmarshals the latest object in the items.

 Actually, we have not found a good method to aggregate the `pod` status. The `PodStatus` can not represent the pod status in multiple clusters figuratively.

Reply:
> This logic looks like it only unmarshals the latest object in the items.
> 
> Actually, we have not found a good method to aggregate the `pod` status. The `PodStatus` can not represent the pod status in multiple clusters figuratively.

This loop only exec once. And aggregating the pod status do not involve multiple clusters.This is just a independent pod resource, not pod in deployment or something else.

Reply:
When we propagate to more than one cluster, there will be more than one item.

Reply:
> When we propagate to more than one cluster, there will be more than one item.

you are right.
So can i set Pod status Ready only when all Pods are in that state?

Reply:
I think it's maybe okay.